src/context/AuthContext.jsx

```diff
@@ -0,0 +1,68 @@
+import axios from "axios";
+import { createContext, useContext, useState } from "react";
+import { useLocation, useNavigate} from "react-router-dom";
+import { Link } from "react-router-dom";
+
+const AuthContext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const navigate = useNavigate();
+  const location = useLocation()
+  const frm = location?.state?.from?.pathname;
+  const [isLogin, setIsLogin] = useState(false);
+  const user = localStorage.getItem("user");
+  const token = localStorage.getItem("token");
+
+  const handleSignupData = async (firstname,lastname,email, password) => {
+    const getValues = { firstname,lastname,email, password};
+    try {
+      const response = await axios.post(`/api/auth/signup`,getValues);
+      console.log(response.status)
+      if (response.status === 200 || response.status === 201) {
+        localStorage.setItem("token", response.data.encodedToken);
+        navigate("/login")
+      }
+    } catch (error) {
+      console.log(error)
+      if (error.response.status === 422) {
+        <div class="alert-items alert-success ">
+              <i class="fas fa-exclamation-triangle alert-icon"></i>Already registered, Please <Link to="/login">Login</Link>
+            </div>
```

Comment:
Instead of manually writing an error message, you can directly extract the exact error message using "error.response.data.errors"